app/src/main/java/com/bashkevich/androidfundamentals/moviesdetails/viewmodel/MoviesDetailsViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package com.bashkevich.androidfundamentals.moviesdetails.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.bashkevich.androidfundamentals.data.JsonLoad
+import com.bashkevich.androidfundamentals.data.Movie
+import kotlinx.coroutines.launch
+
+class MoviesDetailsViewModel(
+        private val jsonLoad: JsonLoad
+) : ViewModel() {
+    private var _movieLiveData  = MutableLiveData<Movie>()
+
+    val movieLiveData : LiveData<Movie>
+    get() = _movieLiveData
+
+    fun getMovieFromList(movieId : Int){
+        viewModelScope.launch {
+            val movies = jsonLoad.loadMovies()
```

Comment:
It would be a bit more efficient if you have a repository that will cache the loaded movie list. But it is optimisation, it works without it as well.